.github/workflows/test.yml

```diff
@@ -11,68 +11,36 @@ env:
-    runs-on: ${{ matrix.operating-system }}
-    strategy:
-      matrix:
-        include:
-          - operating-system: windows-latest
-            vcpkgPackages: Qt5 gdcm boost
-            vcpkgTriplet: 'x64-windows'
-            useVcpkgToolchainFile: true
-            cmakeGenerator: 'VS16Win64'
-          - operating-system: ubuntu-latest            
-            cmakeGenerator: 'Ninja'
-            useVcpkgToolchainFile: false
-        
+    if: >
+      github.repository == 'DataMedSci/gi-yagit' &&
+      !contains(github.event.head_commit.message, '[ci skip]') &&
+      !contains(github.event.head_commit.message, '[skip ci]')
+    name: Build yagit
+    runs-on: ubuntu-latest
+
     steps:
     - uses: actions/checkout@v2
 
-    - name: Install GCC 10 Linux
-      if: ${{ runner.os == 'Linux' }}
-      uses: egor-tensin/setup-gcc@v1
-      with:
-          version: 10
-          platform: x64
-
-    - name: Install Linux requirements
-      if: ${{ runner.os == 'Linux' }}
+    - name: Check versions
       run: |
-           sudo apt update
-           sudo apt install qt5-default libgdcm3.0 libgdcm-dev libboost-all-dev
+        gcc --version
+        g++ --version
+        cmake --version
 
-# Step below not only installs and does VCPKG setup but also installs dependency packages and store them in cache
-# cache is crucial, as installation (compilation from sources) of many dependencies is time consuming
-# i.e. wxwidgets needs >10min for installation and <1min for retrieval from cache
-    - name: Install Windows requirements
-      if: ${{ runner.os == 'Windows' }}
-      uses: lukka/run-vcpkg@v7.3
-      with:
-         vcpkgArguments: '${{ matrix.vcpkgPackages }}'
-         vcpkgTriplet: 'x64-windows'
-         vcpkgDirectory: '${{ runner.workspace }}/b/vcpkg'
-         vcpkgGitCommitId: 'fdb170fda2daa456a843aa3614e5aca6fffa4d52'
+    - name: Install dependencies
+      run: |
+        sudo apt update
+        sudo apt install libgdcm-dev
 
     - name: Configure and build with CMake
-      uses: lukka/run-cmake@v3.4
-      with:
-          vcpkgTriplet: '${{ matrix.vcpkgTriplet }}'  # VCPKG target specification (so called "triplet"), here used only on Windows
-          cmakeBuildType: ${{env.BUILD_TYPE}}  # Release or Debug
-          cmakeGenerator: '${{ matrix.cmakeGenerator }}'  # specify generator (i.e. Ninja or Visual Studio)
-          buildDirectory: 'build'  # subdirectory to store make build and store results
-          buildWithCMake: true  # use CMake to trigger build process
-          useVcpkgToolchainFile: '${{ matrix.useVcpkgToolchainFile }}'  # use vcpkg to install dependencies (i.e. wxwidgets) on Windows
+      working-directory: ${{github.workspace}}/new
+      run: |
+        mkdir build && cd build
+        cmake .. -DCMAKE_BUILD_TYPE=${{env.BUILD_TYPE}} -DBUILD_EXAMPLES=ON
+        cmake --build .
```

Comment:
What do you think of commands which do not require to do `cd build`, like:
```
cmake -S . -B build
cmake --build build
```

?

Reply:
I used version with `cd build`, because it is more popular cmake workflow.
e.g.:
https://cmake.org/cmake/help/latest/guide/tutorial/A%20Basic%20Starting%20Point.html#build-and-run
https://github.com/google/googletest/blob/main/googletest/README.md#build-with-cmake
https://itk.org/ITKSoftwareGuide/html/Book1/ITKSoftwareGuide-Book1ch2.html